### PR TITLE
GEODE-7541: Removing dependency on SocketCreator/Factory

### PR DESCRIPTION
**Pull request by echobravopapa**
- Pushed static call to SocketCreatorFactory up to DistributionImpl
- SocketCreator is now a parameter to MembershipBuilder
- extracted MembershipSocketCreator interface from SocketCreator
- created adapter layer to handle passthru of SocketCreator calls
- renamed with "Tcp" prefix previously refactored socket creator efforts
- Use InetAddress directly from the JDK.

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-core/src/main/java/org/apache/geode/distributed/internal/membership/gms/messenger/JGroupsMessenger.java [hunk 1]
```diff
@@ -537,9 +537,7 @@ private void establishLocalAddress() {
         || !config.getStartLocator().isEmpty();
 
     // establish the DistributedSystem's address
-    String hostname =
```
Comment (bschuchardt) [inline geode-core/src/main/java/org/apache/geode/distributed/internal/membership/gms/messenger/JGroupsMessenger.java]: You'll need to run some rolling upgrade tests with clients having non-durable CQs and servers having enable-network-partition-detection=true to test this change.

File: geode-core/src/main/java/org/apache/geode/distributed/internal/membership/adapter/MembershipSocketCreatorAdapter.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  @Override
+  public Socket connect(final InetAddress inetadd, final int port, final int timeout,
+      final ConnectionWatcher optionalWatcher, final boolean clientSide, final int socketBufferSize,
+      final boolean sslConnection) throws IOException {
+    return socketCreator.connect(inetadd, port, timeout, optionalWatcher, clientSide,
+        socketBufferSize, sslConnection);
+  }
+
+  @Override
+  public boolean shouldResolveDNS() {
```
Comment (bschuchardt) [inline geode-core/src/main/java/org/apache/geode/distributed/internal/membership/adapter/MembershipSocketCreatorAdapter.java]: It looks like you're not using this method

File: geode-core/src/main/java/org/apache/geode/distributed/internal/membership/gms/GMSMemberData.java [hunk 3]
```diff
@@ -95,7 +97,7 @@ public GMSMemberData(String localhost, int udpPort, Version version) {
    * @param p the membership listening port
    */
   @VisibleForTesting
-  public GMSMemberData(String i, int p) {
+  public GMSMemberData(String i, int p) throws UnknownHostException {
```
Comment (upthewaterspout) [inline geode-core/src/main/java/org/apache/geode/distributed/internal/membership/gms/GMSMemberData.java]: I don't see this constructor being used in my intellij project...

Comment (echobravopapa) [reply geode-core/src/main/java/org/apache/geode/distributed/internal/membership/gms/GMSMemberData.java]: removed.

File: geode-core/src/main/java/org/apache/geode/distributed/internal/membership/gms/api/MembershipSocketCreator.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+  ServerSocket createServerSocketUsingPortRange(InetAddress ba, int backlog,
+      boolean isBindAddress, boolean useNIO, int tcpBufferSize, int[] tcpPortRange,
+      boolean sslConnection) throws IOException;
+
+  Socket connect(InetAddress inetadd, int port, int timeout,
+      ConnectionWatcher optionalWatcher, boolean clientSide, int socketBufferSize,
+      boolean sslConnection) throws IOException;
+
+  boolean shouldResolveDNS();
```
[the remark this answers is not in the document]
Comment (echobravopapa) [reply geode-core/src/main/java/org/apache/geode/distributed/internal/membership/gms/api/MembershipSocketCreator.java]: removed.

File: geode-core/src/main/java/org/apache/geode/distributed/internal/membership/gms/messenger/JGroupsMessenger.java [hunk 5]
```diff
@@ -270,7 +270,7 @@ public void init(Services<ID> s) {
     // JGroups UDP protocol requires a bind address
     if (str == null || str.length() == 0) {
       try {
-        str = SocketCreator.getLocalHost().getHostAddress();
+        str = InetAddress.getLocalHost().getHostAddress();
```
[the remark this answers is not in the document]
Comment (echobravopapa) [reply geode-core/src/main/java/org/apache/geode/distributed/internal/membership/gms/messenger/JGroupsMessenger.java]: we'd need to pass in the socket creator to this class, which could be done, wasn't sure it was necessary

File: geode-core/src/main/java/org/apache/geode/distributed/internal/membership/gms/api/MembershipSocketCreator.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.net.UnknownHostException;
+
+import org.apache.geode.distributed.internal.tcpserver.ConnectionWatcher;
+
+public interface MembershipSocketCreator {
```
[the remark this answers is not in the document]
Comment (echobravopapa) [reply geode-core/src/main/java/org/apache/geode/distributed/internal/membership/gms/api/MembershipSocketCreator.java]: will look at this after kicking off some test runs

File: geode-core/src/main/java/org/apache/geode/distributed/internal/membership/gms/messenger/JGroupsMessenger.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (echobravopapa) [reply geode-core/src/main/java/org/apache/geode/distributed/internal/membership/gms/messenger/JGroupsMessenger.java]: testing against this change now

File: geode-core/src/main/java/org/apache/geode/distributed/internal/membership/gms/api/MembershipSocketCreator.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (echobravopapa) [reply geode-core/src/main/java/org/apache/geode/distributed/internal/membership/gms/api/MembershipSocketCreator.java]: this is consolidated now

**Comment by upthewaterspout** [comment]
Changes look ok, but it looks like you have some name resolution related CI failures. So something is definitely broken.

**Pull request merged**